### ISSUE TITLE
object detection API can not set batch size > 1 for R-FCN training

### ISSUE THREAD

**Issue by luowei1107** [tensorflow/models#4831]
when I use object detection API for R-FCN training, I set batch size greater than 1. An error occur: "Cannot batch tensors with different shapes in component 1. First element had shape [600,750,3] and element 1 had shape [901,600,3]."

How can I modify code to support batch size greater than 1? Please help!

**Comment by varun19299** [comment]
That's probably because of a `keep_aspect_ratio_resizer`. That allows for dynamic sizes, hence no batching is permitted.

Use a `fixed_shape_resizer` instead.

**Comment by c329686183** [comment]
I have the same issue, have u solved it ?

**Comment by xDooooot** [comment]
Thank you for your sloution. It helps me a lot, I can use batch_size 16 for training now. 
@varun19299

**Comment by luowei1107** [comment]
Thanks. I modify the configs which image_resizer{ keep_aspect_ratio_resizer{min_dimension: 600max_dimension: 1024 }}  to image_resizer{ fixed_shape_resizer{height: 600 width: 1024 } to solve it.
@varun19299 @c329686183

**Comment by amankhandelia** [comment]
@varun19299 
I have this query, considering the fact that resizing imaging without respecting the aspect ratio will distort the image, will this impact the performance of the model, especially at the time inference (please consider with respect to Mask RCNN)

**Comment by varun19299** [comment]
Yes, that’s a drawback, but you could probably use another pre-processing step where you could resize with padding / another strategy. Regardless there will be a distortion.

**Comment by veonua** [comment]
@varun19299 could you please tell more about pre-processing steps. is it part of the config, or it must be done in step where we create record files.

**Comment by andretkj1** [comment]
> That's probably because of a `keep_aspect_ratio_resizer`. That allows for dynamic sizes, hence no batching is permitted.
> 
> Use a `fixed_shape_resizer` instead.

my config file already use fixed_shape_resizer {
        height: 640
        width: 640}
but still can change my batch size other then 1. when i change to other then 1 i got omm error. i am using faster_rcnn_inception_resnet_v2_640x640_coco17_tpu-8.
